once_test.py

```diff
@@ -155,6 +254,30 @@ def test_different_args_same_result(self):
         self.assertEqual(counting_fn(2), 1)
         self.assertEqual(counter.value, 1)
 
+    def test_partial(self):
+        counter = Counter()
+        func = once.once(functools.partial(lambda _: counter.get_incremented(), None))
+        self.assertEqual(func(), 1)
+        self.assertEqual(func(), 1)
+
+    def test_failing_function(self):
```

Comment:
This test is really counter-intuitive to me. So errors cause us to re-run the function? If so, we should probably do the same thing for our generators, and make sure this is well documented in our API.

Reply:
Added for sync generators, and a TODO to add to async after refactor, and added detailed description in the docstring.

Reply:
https://github.com/DelfinaCare/once/issues/12#issuecomment-1764883796 for reference